aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -213,17 +330,52 @@ module aptos_framework::coin {
         account_addr: address,
         amount: u64,
         dst_coin: &mut AggregatableCoin<CoinType>,
-    ) acquires CoinStore {
+    ) acquires CoinStore, CoinConversionMap, CoinInfo, PairedCoinType {
         // Skip collecting if amount is zero.
         if (amount == 0) {
             return
         };
 
-        let coin_store = borrow_global_mut<CoinStore<CoinType>>(account_addr);
-        let coin = extract(&mut coin_store.coin, amount);
+        let coin_balance = coin_balance<CoinType>(account_addr);
+        let (coin_amount_to_collect, fa_amount_to_collect) = if (coin_balance >= amount) {(amount, 0)} else {(coin_balance, amount - coin_balance)};
+        let coin = zero();
+        if (coin_amount_to_collect > 0) {
+            let coin_store = borrow_global_mut<CoinStore<CoinType>>(account_addr);
+            merge(&mut coin, extract(&mut coin_store.coin, coin_amount_to_collect));
+        };
+        if (fa_amount_to_collect > 0) {
+            let store = primary_fungible_store::primary_store(account_addr, paired_metadata<CoinType>());
+            let fa = fungible_asset::withdraw_internal(object::object_address(&store), fa_amount_to_collect);
+            merge(&mut coin, fungible_asset_to_coin<CoinType>(fa));
+        };
         merge_aggregatable_coin(dst_coin, coin);
     }
 
+    fun maybe_convert_to_fungible_store<CoinType>(
+        account: address
+    ) acquires CoinStore, CoinConversionMap, CoinInfo {
+        if (exists<CoinStore<CoinType>>(account)) {
+            let CoinStore<CoinType> {
+                coin,
+                frozen,
+                deposit_events,
+                withdraw_events
+            } = move_from<CoinStore<CoinType>>(account);
+            event::destory_handle(deposit_events);
+            event::destory_handle(withdraw_events);
```

Comment:
Anyway to store this under their account or maybe in the new fungible asset structs so they have unified event history?

Deposit and withdraw events can be important for accounting purposes e.g. taxable events, among other things

Erasing history seems like an unfavorable practice

Reply:
I would migrate this to event v2 before land this pr so I guess those v2 events would be enough?

Reply:
I'm not entirely sure about what implementation you have in mind, though I think that while v2 events could be preserved, v1 events would not not be: because this operation is destroying a v1 handle, which effectively erases the pointer to the v1 event list

Perhaps during the handle delete operation you emit a v2 style event of the form

```
struct CoinHandleDeletionEvent {
    deleted_deposit_event_handle_creation_num: u64,
    deleted_withdraw_event_handle_creation_num: u64,
} 
```

?

With this event on record, a user can then get their v1 coin events from the events by event handle API

Reply:
Sorry for the confusion. What I mean is: 
 We will emit v2 events for both `coin` and `fungible asset`. So even if I delete the handle for coin module, v2 events for coin will still be there since the txn that updates the framework. 

Do you mean to track the history we have to specify in an event that since this version, v1 is gone?